packages/edit-site/src/components/dataviews/README.md

```diff
@@ -44,8 +44,8 @@ Example:
 	},
 	search: '',
 	filters: {
-		author: 2,
-		status: 'publish, draft'
+		author: { in: 2 },
+		status: { in: 'publish, draft' }
```

Comment:
I feel a better representation of the filter is something like `{ operator: 'in', values: 2 }` because in a lot of cases you should be able to switch the operator without switching the values.

Reply:
I think we need to support multiple operators per field and being ready for logical operators (and/or). This is how it looks like in the current approach:

```js
{
  filters: {
    or: {
      date: { before: "20231001", after: "20230101" },
    }
  }
}
```

Reply:
I actually prefer something like

```
filters: {
   operator: 'or',
   children: [
      { 'field': 'author', operator: 'startsWith', value: "something" ],
      { 'field': 'author', operator: 'contain', value: "ok" ],
      { 'field2': 'author', operator: 'in', values: ["blabla" ] },
   ]
}
```

Reply:
which could also support this:

```
filters: {
   operator: 'or',
   children: [
      { 'field': 'author', operator: 'startsWith', value: "something" ],
      { 'field': 'author', operator: 'contain', value: "ok" ],
      { 'field2': 'author', operator: 'in', values: ["blabla" ] },
      { operator: 'and', children: [
         { 'field': 'status', operator: 'in', values: [ 'draft', 'publish' ] }
      ] }
   ]
}
```

Reply:
Do you feel strongly about that filters shape? I don't see it being easier to work with for changing operators. It's also a lot more verbose. Example:

```js
{
  filters: [
    { field: 'author', operator: 'in', value: 2 },
  ]
}
```

Reply:
I don't feel strongly but I think it's actually easier to work with compared to what you propose:

 - A filter object is self sufficient, it contains all the information it needs (the field id, the operator and the values), it's not on multiple levels.
 - Filter objets can be combined.

So basically you can pass a filter object to a component and renders it entirely. 

---
In your proposal, it's not clear to me, how you handle: multiple operators for the same field and nested logical operators. Can you translate the example I have here in your format to get more clarity https://github.com/WordPress/gutenberg/pull/55735#discussion_r1379821822 ?

Reply:
> In your proposal, it's not clear to me, how you handle: multiple operators for the same field and nested logical operators. Can you translate the example I have here in your format to get more clarity https://github.com/WordPress/gutenberg/pull/55735#discussion_r1379821822 ?

I already did right at the beginning :) 

I went ahead and implemented your proposal at https://github.com/WordPress/gutenberg/pull/55735/commits/379fae15704d9865812c6e39e16d979e49e7d07f

Reply:
> I already did right at the beginning of this thread and in the PR description :)

I see the multiple operators but not the "and" nested within the "or"